test/lib/buildFactory.test.js

```diff
@@ -1005,4 +1007,175 @@ describe('Build Factory', () => {
             );
         });
     });
+
+    describe('getBuildStatuses', () => {
+        let config;
+        let expected;
+        let returnValue;
+        let queryConfig;
+
+        beforeEach(() => {
+            sinon.stub(BuildFactory.prototype, 'query').returns();
+
+            config = {
+                jobIds: [1, 2, 3, 4],
+                offset: 1,
+                numBuilds: 5
+            };
+
+            returnValue = [
+                [
```

Comment:
also is this the correct return data type? an array of arrays?

Reply:
@InderH

Reply:
yup, it should be an array of arrays - when running a raw query on sequelize without having it map the returning data to a model (which in this case we don't), it returns an array with 2 elements, the first being an array of the data/rows, and the second being meta data.

Reply:
hm but the expected query config has `table: 'builds'`, right?

Reply:
Yeah it has `table: 'builds'` but also has `rawResponse: true` - we are only getting a small subset of the data a build has and are using the data in a way which casting it to a build model wouldn't help, so having the query response map back to builds model would just be extra unnecessary computation

Reply:
I see
sounds reasonable